### DISCUSSION TITLE
What parameters to launch GAMA 2 ?

### DISCUSSION THREAD

**Discussion by RoiArthurB** [gama-platform/gama#107] [General]
GAMA 2.0 is announced and will change the way GAMA can be launched (with a more modular and flexible architecture). This flexibility will allow launching different kind of GAMA, mostly controlled by some [flags](https://en.wikipedia.org/wiki/Feature_toggle) in the start.

I'm then asking here to the community : *What parameter seems to be the more important to precise when you'll launch GAMA 2.0?*

To give you some example, a flag `server` to work with a headless version of GAMA could be an answer. In a different approach, a flag `presentation` launching one simulation right away.

**Comment by RoiArthurB** [comment]
Ping developers @gama-platform/core @gama-platform/committers 🤗🤗🤗

**Reply by ndgnuh** [reply]
I think this relates more to how the user uses Gama, won't it be more appropriate to ask on user places (e.g. Gama Day slack?)

**Reply by AlexisDrogoul** [reply]
It is true to a certain extent. Advanced users will use this functionality, and they are quite well represented among the developers...

**Reply by RoiArthurB** [reply]
@ndgnuh Good idea, feel free to send the link on the GAMA Days Slack (I lost the link) 😉

**Comment by AlexisDrogoul** [comment]
Hi, maybe we should begin to list the various options existing for both the headless and the UI version of GAMA (https://gama-platform.github.io/wiki/Launching#launching-the-application-from-the-command-line) ? In UI, we have : 

1. The direct use of a model file : launch GAMA and opens an editor on it. 
2. The use of a model file + `#` + experiment name to launch directly an experiment 
3. The use of `-D`+ a preference name or a flag name (https://github.com/gama-platform/gama/blob/GAMA_1.8.2/ummisco.gama.annotations/src/ummisco/gama/dev/utils/FLAGS.java) to set its new value
4. Some things are also supposed to work, but I guess they do not really, like `-clean` to clean the metadata at launch or '-data' to specify a workspace. 

The goal is to have a unified way of launching GAMA from the command line (or a config file, or another language, like R), simple enough for beginners but with enough options to satisfy the geeks and external tooling. 

A second use would be to embed these parameters in the GAML files directly, in the form of pragmas or in experiments, so that model files can be self contained; ie when you launch one using the command above (gama + filename + experiment), GAMA will look for directives on what to do with it: launch it, set some preferences, use x cores, etc... whatever.

**Comment by AlexisDrogoul** [comment]
In any case, I am strongly in favour of using a tool like https://picocli.info for managing these options, producing the documentation, etc.
